docs/user-guide/global-configurations/cluster-and-environments.md

```diff
@@ -1,87 +1,81 @@
+{% endtabs %}
 
-* **Bearer token**
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/cluster-and-environments/generate-cluster-credentials.png)
 
-Provide your kubernetes cluster’s Bearer token for authentication purposes so that Devtron is able to communicate with your kubernetes cluster and can deploy your application in your kubernetes cluster.
+#### Server URL
 
-Generate the **`Bearer Token`** by running the following command: 
+Provide the server URL of your kubernetes cluster. It is recommended to use a self-hosted URL instead of cloud hosted. Self-hosted URL will provide the following benefits:
 
-```bash
-curl -O https://raw.githubusercontent.com/devtron-labs/utilities/main/kubeconfig-exporter/kubernetes_export_sa.sh && bash kubernetes_export_sa.sh cd-user devtroncd https://raw.githubusercontent.com/devtron-labs/utilities/main/kubeconfig-exporter/clusterrole.yaml
-```
+**\(a\) Disaster Recovery -** It is not possible to edit the server-url of a cluster. So if you're using an eks url, For eg- ` *****.eu-west-1.elb.amazonaws.com` it will be a tedious task to add a new cluster and migrate all the services one by one. While using a self-hosted url For eg- `clear.example.com` you can just point to the new cluster's server url in DNS manager and update the new cluster token and sync all the deployments.
 
-Please ensure that kubectl and jq are installed on the bastion on which you are running the command.
+**\(b\) Easy cluster migrations -** Cluster url is given in the name of the cloud provider used, so migrating your cluster from one provider to another will result in waste of time and effort. On the other hand, if using a self-hosted url migrations will be easy as the url is of single hosted domain independent of the cloud provider.
 
-![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/cluster-and-environments/bearer-token.png)
+#### Bearer token
 
-If you are using a **`microk8s cluster`**, run the following command to generate the bearer token:
+Provide your kubernetes cluster’s Bearer token for authentication purposes so that Devtron is able to communicate with your kubernetes cluster and can deploy your application in your kubernetes cluster.
 
-```bash
-curl -O https://raw.githubusercontent.com/devtron-labs/utilities/main/kubeconfig-exporter/kubernetes_export_sa.sh && sed -i 's/kubectl/microk8s kubectl/g' kubernetes_export_sa.sh && bash kubernetes_export_sa.sh cd-user devtroncd https://raw.githubusercontent.com/devtron-labs/utilities/main/kubeconfig-exporter/clusterrole.yaml
-```
+### 3. Enable applications metrics for the cluster
 
-### 3. Prometheus Info
+If you want to see application metrics against the applications deployed in the  cluster, Prometheus should be deployed in the cluster. Prometheus is a powerful tool to provide graphical insight into your application behavior. The below inputs are required to configure Prometheus over Devtron.
 
```

Comment:
Note: Make sure that you install Monitoring (Grafana) from the Devtron Stack Manager to configure prometheus.

And the tabular form for the fields.